#### KDD
##### Aditya Ambre
##### 20200802023
##### LAB - 3 Preprocessing Normalization and Standardization

**Aim:** To perform Preprocessing Normalization and Standardization on the given dataset using

1.	By creating functions.
2.	By using the sklearn library.

And validating the results.


In [1]:
# Importing Required libraries
import pandas as pd
import numpy as np
from sklearn.preprocessing import MinMaxScaler
from sklearn.preprocessing import StandardScaler

In [3]:
# Loading dataset
house_df = pd.read_csv("train.csv")
house_df

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0        1          60       RL         65.0     8450   Pave   NaN      Reg   
1        2          20       RL         80.0     9600   Pave   NaN      Reg   
2        3          60       RL         68.0    11250   Pave   NaN      IR1   
3        4          70       RL         60.0     9550   Pave   NaN      IR1   
4        5          60       RL         84.0    14260   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
1455  1456          60       RL         62.0     7917   Pave   NaN      Reg   
1456  1457          20       RL         85.0    13175   Pave   NaN      Reg   
1457  1458          70       RL         66.0     9042   Pave   NaN      Reg   
1458  1459          20       RL         68.0     9717   Pave   NaN      Reg   
1459  1460          20       RL         75.0     9937   Pave   NaN      Reg   

     LandContour Utilities  ... PoolArea PoolQC  Fence MiscFeature MiscVal  \
0            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
2            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
3            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
4            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
...          ...       ...  ...      ...    ...    ...         ...     ...   
1455         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1456         Lvl    AllPub  ...        0    NaN  MnPrv         NaN       0   
1457         Lvl    AllPub  ...        0    NaN  GdPrv        Shed    2500   
1458         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1459         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   

     MoSold YrSold  SaleType  SaleCondition  SalePrice  
0         2   2008        WD         Normal     208500  
1         5   2007        WD         Normal     181500  
2         9   2008        WD         Normal     223500  
3         2   2006        WD        Abnorml     140000  
4        12   2008        WD         Normal     250000  
...     ...    ...       ...            ...        ...  
1455      8   2007        WD         Normal     175000  
1456      2   2010        WD         Normal     210000  
1457      5   2010        WD         Normal     266500  
1458      4   2010        WD         Normal     142125  
1459      6   2008        WD         Normal     147500  

[1460 rows x 81 columns]

In [4]:
house_df = house_df.set_index(["Id"])
house_df

MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
Id                                                                      
1             60       RL         65.0     8450   Pave   NaN      Reg   
2             20       RL         80.0     9600   Pave   NaN      Reg   
3             60       RL         68.0    11250   Pave   NaN      IR1   
4             70       RL         60.0     9550   Pave   NaN      IR1   
5             60       RL         84.0    14260   Pave   NaN      IR1   
...          ...      ...          ...      ...    ...   ...      ...   
1456          60       RL         62.0     7917   Pave   NaN      Reg   
1457          20       RL         85.0    13175   Pave   NaN      Reg   
1458          70       RL         66.0     9042   Pave   NaN      Reg   
1459          20       RL         68.0     9717   Pave   NaN      Reg   
1460          20       RL         75.0     9937   Pave   NaN      Reg   

     LandContour Utilities LotConfig  ... PoolArea PoolQC  Fence MiscFeature  \
Id                                    ...                                      
1            Lvl    AllPub    Inside  ...        0    NaN    NaN         NaN   
2            Lvl    AllPub       FR2  ...        0    NaN    NaN         NaN   
3            Lvl    AllPub    Inside  ...        0    NaN    NaN         NaN   
4            Lvl    AllPub    Corner  ...        0    NaN    NaN         NaN   
5            Lvl    AllPub       FR2  ...        0    NaN    NaN         NaN   
...          ...       ...       ...  ...      ...    ...    ...         ...   
1456         Lvl    AllPub    Inside  ...        0    NaN    NaN         NaN   
1457         Lvl    AllPub    Inside  ...        0    NaN  MnPrv         NaN   
1458         Lvl    AllPub    Inside  ...        0    NaN  GdPrv        Shed   
1459         Lvl    AllPub    Inside  ...        0    NaN    NaN         NaN   
1460         Lvl    AllPub    Inside  ...        0    NaN    NaN         NaN   

     MiscVal MoSold  YrSold  SaleType  SaleCondition  SalePrice  
Id                                                               
1          0      2    2008        WD         Normal     208500  
2          0      5    2007        WD         Normal     181500  
3          0      9    2008        WD         Normal     223500  
4          0      2    2006        WD        Abnorml     140000  
5          0     12    2008        WD         Normal     250000  
...      ...    ...     ...       ...            ...        ...  
1456       0      8    2007        WD         Normal     175000  
1457       0      2    2010        WD         Normal     210000  
1458    2500      5    2010        WD         Normal     266500  
1459       0      4    2010        WD         Normal     142125  
1460       0      6    2008        WD         Normal     147500  

[1460 rows x 80 columns]

In [25]:
all_data_na = (house_df.isnull().sum() / len(house_df)) * 100
all_data_na = all_data_na.drop(all_data_na[all_data_na == 0].index).sort_values(ascending = True)
missing_data = pd.DataFrame({'Missing Ratio (in %)':all_data_na})
missing_data.head(20)


Empty DataFrame
Columns: [Missing Ratio (in %)]
Index: []

In [8]:
house_df = house_df.drop(['PoolQC', 'MiscFeature','Alley','Fence','FireplaceQu'], axis=1)
house_df

MSSubClass MSZoning  LotFrontage  LotArea Street LotShape LandContour  \
Id                                                                            
1             60       RL         65.0     8450   Pave      Reg         Lvl   
2             20       RL         80.0     9600   Pave      Reg         Lvl   
3             60       RL         68.0    11250   Pave      IR1         Lvl   
4             70       RL         60.0     9550   Pave      IR1         Lvl   
5             60       RL         84.0    14260   Pave      IR1         Lvl   
...          ...      ...          ...      ...    ...      ...         ...   
1456          60       RL         62.0     7917   Pave      Reg         Lvl   
1457          20       RL         85.0    13175   Pave      Reg         Lvl   
1458          70       RL         66.0     9042   Pave      Reg         Lvl   
1459          20       RL         68.0     9717   Pave      Reg         Lvl   
1460          20       RL         75.0     9937   Pave      Reg         Lvl   

     Utilities LotConfig LandSlope  ... EnclosedPorch 3SsnPorch ScreenPorch  \
Id                                  ...                                       
1       AllPub    Inside       Gtl  ...             0         0           0   
2       AllPub       FR2       Gtl  ...             0         0           0   
3       AllPub    Inside       Gtl  ...             0         0           0   
4       AllPub    Corner       Gtl  ...           272         0           0   
5       AllPub       FR2       Gtl  ...             0         0           0   
...        ...       ...       ...  ...           ...       ...         ...   
1456    AllPub    Inside       Gtl  ...             0         0           0   
1457    AllPub    Inside       Gtl  ...             0         0           0   
1458    AllPub    Inside       Gtl  ...             0         0           0   
1459    AllPub    Inside       Gtl  ...           112         0           0   
1460    AllPub    Inside       Gtl  ...             0         0           0   

     PoolArea MiscVal  MoSold  YrSold  SaleType  SaleCondition SalePrice  
Id                                                                        
1           0       0       2    2008        WD         Normal    208500  
2           0       0       5    2007        WD         Normal    181500  
3           0       0       9    2008        WD         Normal    223500  
4           0       0       2    2006        WD        Abnorml    140000  
5           0       0      12    2008        WD         Normal    250000  
...       ...     ...     ...     ...       ...            ...       ...  
1456        0       0       8    2007        WD         Normal    175000  
1457        0       0       2    2010        WD         Normal    210000  
1458        0    2500       5    2010        WD         Normal    266500  
1459        0       0       4    2010        WD         Normal    142125  
1460        0       0       6    2008        WD         Normal    147500  

[1460 rows x 75 columns]

In [9]:
a = house_df.isnull().sum().sort_values(ascending=False)[:19].to_frame()
l = list(a.index.values)
house_df[l].info()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 1460 entries, 1 to 1460
Data columns (total 19 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   LotFrontage   1201 non-null   float64
 1   GarageType    1379 non-null   object 
 2   GarageYrBlt   1379 non-null   float64
 3   GarageFinish  1379 non-null   object 
 4   GarageQual    1379 non-null   object 
 5   GarageCond    1379 non-null   object 
 6   BsmtFinType2  1422 non-null   object 
 7   BsmtExposure  1422 non-null   object 
 8   BsmtQual      1423 non-null   object 
 9   BsmtCond      1423 non-null   object 
 10  BsmtFinType1  1423 non-null   object 
 11  MasVnrArea    1452 non-null   float64
 12  MasVnrType    1452 non-null   object 
 13  Electrical    1459 non-null   object 
 14  KitchenAbvGr  1460 non-null   int64  
 15  BedroomAbvGr  1460 non-null   int64  
 16  HalfBath      1460 non-null   int64  
 17  FullBath      1460 non-null   int64  
 18  BsmtHalfBath  1460 non-null 

In [10]:
num_columns = list(house_df[l].select_dtypes(include=np.number).columns)
cat_columns = list(house_df[l].select_dtypes(exclude=np.number).columns)

In [12]:
print("Variables : ", num_columns)
print("Categorical Variables : ",cat_columns)

Variables :  ['LotFrontage', 'GarageYrBlt', 'MasVnrArea', 'KitchenAbvGr', 'BedroomAbvGr', 'HalfBath', 'FullBath', 'BsmtHalfBath']
Categorical Variables :  ['GarageType', 'GarageFinish', 'GarageQual', 'GarageCond', 'BsmtFinType2', 'BsmtExposure', 'BsmtQual', 'BsmtCond', 'BsmtFinType1', 'MasVnrType', 'Electrical']


In [13]:
for i in list(house_df[l].columns):
  if i in num_columns:
    house_df[i].fillna(house_df[i].median(), inplace=True)
  else:
    house_df[i].fillna(house_df[i].mode()[0], inplace=True)


In [24]:
from numpy.core.multiarray import asanyarray
all_data_na = (house_df.isnull().sum() / len(house_df)) * 100
all_data_na = all_data_na.drop(all_data_na[all_data_na == 0].index).sort_values(ascending = True)
missing_data = pd.DataFrame({'Missing Ratio (in %)':all_data_na})
missing_data


Empty DataFrame
Columns: [Missing Ratio (in %)]
Index: []

In [14]:
house_df

MSSubClass MSZoning  LotFrontage  LotArea Street LotShape LandContour  \
Id                                                                            
1             60       RL         65.0     8450   Pave      Reg         Lvl   
2             20       RL         80.0     9600   Pave      Reg         Lvl   
3             60       RL         68.0    11250   Pave      IR1         Lvl   
4             70       RL         60.0     9550   Pave      IR1         Lvl   
5             60       RL         84.0    14260   Pave      IR1         Lvl   
...          ...      ...          ...      ...    ...      ...         ...   
1456          60       RL         62.0     7917   Pave      Reg         Lvl   
1457          20       RL         85.0    13175   Pave      Reg         Lvl   
1458          70       RL         66.0     9042   Pave      Reg         Lvl   
1459          20       RL         68.0     9717   Pave      Reg         Lvl   
1460          20       RL         75.0     9937   Pave      Reg         Lvl   

     Utilities LotConfig LandSlope  ... EnclosedPorch 3SsnPorch ScreenPorch  \
Id                                  ...                                       
1       AllPub    Inside       Gtl  ...             0         0           0   
2       AllPub       FR2       Gtl  ...             0         0           0   
3       AllPub    Inside       Gtl  ...             0         0           0   
4       AllPub    Corner       Gtl  ...           272         0           0   
5       AllPub       FR2       Gtl  ...             0         0           0   
...        ...       ...       ...  ...           ...       ...         ...   
1456    AllPub    Inside       Gtl  ...             0         0           0   
1457    AllPub    Inside       Gtl  ...             0         0           0   
1458    AllPub    Inside       Gtl  ...             0         0           0   
1459    AllPub    Inside       Gtl  ...           112         0           0   
1460    AllPub    Inside       Gtl  ...             0         0           0   

     PoolArea MiscVal  MoSold  YrSold  SaleType  SaleCondition SalePrice  
Id                                                                        
1           0       0       2    2008        WD         Normal    208500  
2           0       0       5    2007        WD         Normal    181500  
3           0       0       9    2008        WD         Normal    223500  
4           0       0       2    2006        WD        Abnorml    140000  
5           0       0      12    2008        WD         Normal    250000  
...       ...     ...     ...     ...       ...            ...       ...  
1456        0       0       8    2007        WD         Normal    175000  
1457        0       0       2    2010        WD         Normal    210000  
1458        0    2500       5    2010        WD         Normal    266500  
1459        0       0       4    2010        WD         Normal    142125  
1460        0       0       6    2008        WD         Normal    147500  

[1460 rows x 75 columns]

In [15]:
num_columns = list(house_df.select_dtypes(include=np.number).columns)
numeric_house_df = house_df[num_columns]
numeric_house_df

MSSubClass  LotFrontage  LotArea  OverallQual  OverallCond  YearBuilt  \
Id                                                                            
1             60         65.0     8450            7            5       2003   
2             20         80.0     9600            6            8       1976   
3             60         68.0    11250            7            5       2001   
4             70         60.0     9550            7            5       1915   
5             60         84.0    14260            8            5       2000   
...          ...          ...      ...          ...          ...        ...   
1456          60         62.0     7917            6            5       1999   
1457          20         85.0    13175            6            6       1978   
1458          70         66.0     9042            7            9       1941   
1459          20         68.0     9717            5            6       1950   
1460          20         75.0     9937            5            6       1965   

      YearRemodAdd  MasVnrArea  BsmtFinSF1  BsmtFinSF2  ...  WoodDeckSF  \
Id                                                      ...               
1             2003       196.0         706           0  ...           0   
2             1976         0.0         978           0  ...         298   
3             2002       162.0         486           0  ...           0   
4             1970         0.0         216           0  ...           0   
5             2000       350.0         655           0  ...         192   
...            ...         ...         ...         ...  ...         ...   
1456          2000         0.0           0           0  ...           0   
1457          1988       119.0         790         163  ...         349   
1458          2006         0.0         275           0  ...           0   
1459          1996         0.0          49        1029  ...         366   
1460          1965         0.0         830         290  ...         736   

      OpenPorchSF  EnclosedPorch  3SsnPorch  ScreenPorch  PoolArea  MiscVal  \
Id                                                                            
1              61              0          0            0         0        0   
2               0              0          0            0         0        0   
3              42              0          0            0         0        0   
4              35            272          0            0         0        0   
5              84              0          0            0         0        0   
...           ...            ...        ...          ...       ...      ...   
1456           40              0          0            0         0        0   
1457            0              0          0            0         0        0   
1458           60              0          0            0         0     2500   
1459            0            112          0            0         0        0   
1460           68              0          0            0         0        0   

      MoSold  YrSold  SalePrice  
Id                               
1          2    2008     208500  
2          5    2007     181500  
3          9    2008     223500  
4          2    2006     140000  
5         12    2008     250000  
...      ...     ...        ...  
1456       8    2007     175000  
1457       2    2010     210000  
1458       5    2010     266500  
1459       4    2010     142125  
1460       6    2008     147500  

[1460 rows x 37 columns]

In [ ]:
#LOG SCALING

def log_scaling(df):
  new = df
  for i in list(new.columns):
    new[i] = np.log(df[i])
    return new

california_log_scaling = log_scaling(numeric_house_df)
california_log_scaling


In [ ]:
# Min-Max
def min_max(df):
  new = df
  for i in list(new.columns):
    new[i] = (new[i] - min(new[i]))/(max(new[i]) - min(new[i]))
    return new

california_min_max_udf = min_max(numeric_house_df)
california_min_max_udf


In [18]:
# validating Min-Max
min_max_scaler = MinMaxScaler()
california_min_max_sklearn = pd.DataFrame()
l = list(numeric_house_df.columns)
california_min_max_sklearn[l] = min_max_scaler.fit_transform(numeric_house_df[l])

california_min_max_sklearn


MSSubClass  LotFrontage   LotArea  OverallQual  OverallCond  YearBuilt  \
0       0.487992     0.150685  0.033420     0.666667        0.500   0.949275   
1       0.000000     0.202055  0.038795     0.555556        0.875   0.753623   
2       0.487992     0.160959  0.046507     0.666667        0.500   0.934783   
3       0.556464     0.133562  0.038561     0.666667        0.500   0.311594   
4       0.487992     0.215753  0.060576     0.777778        0.500   0.927536   
...          ...          ...       ...          ...          ...        ...   
1455    0.487992     0.140411  0.030929     0.555556        0.500   0.920290   
1456    0.000000     0.219178  0.055505     0.555556        0.625   0.768116   
1457    0.556464     0.154110  0.036187     0.666667        1.000   0.500000   
1458    0.000000     0.160959  0.039342     0.444444        0.625   0.565217   
1459    0.000000     0.184932  0.040370     0.444444        0.625   0.673913   

      YearRemodAdd  MasVnrArea  BsmtFinSF1  BsmtFinSF2  ...  WoodDeckSF  \
0         0.883333    0.122500    0.125089    0.000000  ...    0.000000   
1         0.433333    0.000000    0.173281    0.000000  ...    0.347725   
2         0.866667    0.101250    0.086109    0.000000  ...    0.000000   
3         0.333333    0.000000    0.038271    0.000000  ...    0.000000   
4         0.833333    0.218750    0.116052    0.000000  ...    0.224037   
...            ...         ...         ...         ...  ...         ...   
1455      0.833333    0.000000    0.000000    0.000000  ...    0.000000   
1456      0.633333    0.074375    0.139972    0.110583  ...    0.407235   
1457      0.933333    0.000000    0.048724    0.000000  ...    0.000000   
1458      0.766667    0.000000    0.008682    0.698100  ...    0.427071   
1459      0.250000    0.000000    0.147059    0.196744  ...    0.858810   

      OpenPorchSF  EnclosedPorch  3SsnPorch  ScreenPorch  PoolArea  MiscVal  \
0        0.111517       0.000000        0.0          0.0       0.0  0.00000   
1        0.000000       0.000000        0.0          0.0       0.0  0.00000   
2        0.076782       0.000000        0.0          0.0       0.0  0.00000   
3        0.063985       0.492754        0.0          0.0       0.0  0.00000   
4        0.153565       0.000000        0.0          0.0       0.0  0.00000   
...           ...            ...        ...          ...       ...      ...   
1455     0.073126       0.000000        0.0          0.0       0.0  0.00000   
1456     0.000000       0.000000        0.0          0.0       0.0  0.00000   
1457     0.109689       0.000000        0.0          0.0       0.0  0.16129   
1458     0.000000       0.202899        0.0          0.0       0.0  0.00000   
1459     0.124314       0.000000        0.0          0.0       0.0  0.00000   

        MoSold  YrSold  SalePrice  
0     0.090909    0.50   0.241078  
1     0.363636    0.25   0.203583  
2     0.727273    0.50   0.261908  
3     0.090909    0.00   0.145952  
4     1.000000    0.50   0.298709  
...        ...     ...        ...  
1455  0.636364    0.25   0.194556  
1456  0.090909    1.00   0.243161  
1457  0.363636    1.00   0.321622  
1458  0.272727    1.00   0.148903  
1459  0.454545    0.50   0.156367  

[1460 rows x 37 columns]

In [ ]:
# Z-Score
def z_scores(df):
  new = df
  for i in list(new.columns):
    new[i] = (new[i] - np.mean(new[i]))/(np.std(new[i]))
    return new

california_z_score_udf = z_scores(numeric_house_df)
california_z_score_udf


In [20]:
# Validating z-score
z_score = StandardScaler()
california_z_scores_sklearn = pd.DataFrame()
l = list(numeric_house_df.columns)
california_z_scores_sklearn[l] = z_score.fit_transform(numeric_house_df[l])

california_z_scores_sklearn


MSSubClass  LotFrontage   LotArea  OverallQual  OverallCond  YearBuilt  \
0       0.430516    -0.220875 -0.207142     0.651479    -0.517200   1.050994   
1      -1.128983     0.460320 -0.091886    -0.071836     2.179628   0.156734   
2       0.430516    -0.084636  0.073480     0.651479    -0.517200   0.984752   
3       0.649335    -0.447940 -0.096897     0.651479    -0.517200  -1.863632   
4       0.430516     0.641972  0.375148     1.374795    -0.517200   0.951632   
...          ...          ...       ...          ...          ...        ...   
1455    0.430516    -0.357114 -0.260560    -0.071836    -0.517200   0.918511   
1456   -1.128983     0.687385  0.266407    -0.071836     0.381743   0.222975   
1457    0.649335    -0.175462 -0.147810     0.651479     3.078570  -1.002492   
1458   -1.128983    -0.084636 -0.080160    -0.795151     0.381743  -0.704406   
1459   -1.128983     0.233255 -0.058112    -0.795151     0.381743  -0.207594   

      YearRemodAdd  MasVnrArea  BsmtFinSF1  BsmtFinSF2  ...  WoodDeckSF  \
0         0.878668    0.514104    0.575425   -0.288653  ...   -0.752176   
1        -0.429577   -0.570750    1.171992   -0.288653  ...    1.626195   
2         0.830215    0.325915    0.092907   -0.288653  ...   -0.752176   
3        -0.720298   -0.570750   -0.499274   -0.288653  ...   -0.752176   
4         0.733308    1.366489    0.463568   -0.288653  ...    0.780197   
...            ...         ...         ...         ...  ...         ...   
1455      0.733308   -0.570750   -0.973018   -0.288653  ...   -0.752176   
1456      0.151865    0.087911    0.759659    0.722112  ...    2.033231   
1457      1.024029   -0.570750   -0.369871   -0.288653  ...   -0.752176   
1458      0.539493   -0.570750   -0.865548    6.092188  ...    2.168910   
1459     -0.962566   -0.570750    0.847389    1.509640  ...    5.121921   

      OpenPorchSF  EnclosedPorch  3SsnPorch  ScreenPorch  PoolArea   MiscVal  \
0        0.216503      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   
1       -0.704483      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   
2       -0.070361      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   
3       -0.176048       4.092524  -0.116339    -0.270208 -0.068692 -0.087688   
4        0.563760      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   
...           ...            ...        ...          ...       ...       ...   
1455    -0.100558      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   
1456    -0.704483      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   
1457     0.201405      -0.359325  -0.116339    -0.270208 -0.068692  4.953112   
1458    -0.704483       1.473789  -0.116339    -0.270208 -0.068692 -0.087688   
1459     0.322190      -0.359325  -0.116339    -0.270208 -0.068692 -0.087688   

        MoSold    YrSold  SalePrice  
0    -1.599111  0.138777   0.347273  
1    -0.489110 -0.614439   0.007288  
2     0.990891  0.138777   0.536154  
3    -1.599111 -1.367655  -0.515281  
4     2.100892  0.138777   0.869843  
...        ...       ...        ...  
1455  0.620891 -0.614439  -0.074560  
1456 -1.599111  1.645210   0.366161  
1457 -0.489110  1.645210   1.077611  
1458 -0.859110  1.645210  -0.488523  
1459 -0.119110  0.138777  -0.420841  

[1460 rows x 37 columns]

# **Observation**

1. Diemensions of the Data set have been greatly reduced to  <b>1460 X 37</b>
2. LotFrontage  has the least number of Non Null values <b>1201</b>
3. During Data Preprocessing we have reduce <b>38</b> non numerical Data Coloumns.

##### Conclusion : Data preprocessing , Normalization and Standardization have been perfomed successfully on the Houseprices Dataset.